smtp.go

```diff
@@ -2,209 +2,334 @@ package maddy
 
 import (
 	"bytes"
+	"crypto/tls"
 	"errors"
 	"fmt"
 	"io"
+	"log"
+	"net"
 	"os"
-	"strings"
-	"time"
-
-	"github.com/emersion/go-msgauth"
-	pgppubkey "github.com/emersion/go-pgp-pubkey"
-	pgppubkeylocal "github.com/emersion/go-pgp-pubkey/local"
-	pgplocal "github.com/emersion/go-pgpmail/local"
-	smtppgp "github.com/emersion/go-pgpmail/smtp"
+	"strconv"
+
 	"github.com/emersion/go-smtp"
-	smtpproxy "github.com/emersion/go-smtp-proxy"
-	"github.com/emersion/go-smtp/backendutil"
-	"github.com/mholt/caddy/caddyfile"
-	"golang.org/x/crypto/openpgp"
+	"github.com/emersion/maddy/config"
+	"github.com/emersion/maddy/module"
 )
 
-func newSMTPServer(tokens map[string][]caddyfile.Token, hostname string, lmtp bool) (server, error) {
-	be, err := newUpstream(tokens)
-	if err != nil {
-		return nil, err
-	}
+type SMTPUser struct {
+	endp      *SMTPEndpoint
+	anonymous bool
+	username  string
+	state     *smtp.ConnectionState
+}
 
-	if tokens, ok := tokens["auth"]; ok {
-		var err error
-		be, err = newSMTPAuth(caddyfile.NewDispenserTokens("", tokens), be)
-		if err != nil {
-			return nil, err
+func (u SMTPUser) Send(from string, to []string, r io.Reader) error {
+	ctx := module.DeliveryContext{}
+	ctx.From = from
```

Comment:
Can't this be initialized inside `module.DeliveryContext{}` braces?

Reply:
Ok.